core/shared/src/test/scala/fs2/concurrent/ChannelSuite.scala

```diff
@@ -251,4 +251,89 @@ class ChannelSuite extends Fs2Suite {
 
     TestControl.executeEmbed(test)
   }
+
+  test("sendAndClose closes right after sending the last element (bounded case)") {
+    val result = Channel.bounded[IO, Int](1).flatMap { ch =>
+      ch.sendAndClose(0) *> ch.isClosed
+    }
+
+    result.assertEquals(true)
+  }
+
+  test("sendAndClose closes right after sending the last element (unbounded case)") {
+    val result = Channel.unbounded[IO, Int].flatMap { ch =>
+      ch.sendAndClose(0) *> ch.isClosed
+    }
+
+    result.assertEquals(true)
+  }
+
+  test("racing send and sendAndClose should work in bounded(1) case") {
```

Comment:
Instead of repeating this code four times, can we extract it to a function and test it with different `Channel`s?

Also, because the test is non-deterministic we should run it multiple times with `replicateA_(100)` or something.